src/NuGet.Core/NuGet.Build.Tasks/Common/MSBuildLogger.cs

```diff
@@ -83,7 +83,7 @@ public override void Log(ILogMessage message)
             }
         }
 
-        private void Log(RestoreLogMessage message)
+        private void Log(IRestoreLogMessage message)
```

Comment:
So because ``logMessage`` above is an ``IRestoreLogMessage`` it was going to ``ILogMessage`` instead and looping?

Maybe it would be safer to just name this method something else such as, ``WriteToMSBuild``